### PR TITLE
[energidataservice] Provide JavaScript example for persisting total price

### PR DESCRIPTION
**Pull request by jlaur**
Unfortunately I can only provide this example in JavaScript, since I have not been successful in figuring out how to use [PersistenceExtensions](https://www.openhab.org/javadoc/latest/org/openhab/core/persistence/extensions/persistenceextensions) methods from DSL.

### CODE REVIEW

File: bundles/org.openhab.binding.energidataservice/README.md [hunk 1]
```diff
@@ -69,6 +71,42 @@ This has the following advantages:
...
+
+:::: tabs
+
+::: tab JavaScript
+
+```javascript
+rules.when()
+    .channel('energidataservice:service:energidataservice:electricity#event').triggered('DAY_AHEAD_AVAILABLE')
+	.then(event =>
+	{
```
Comment (florian-h05) [inline bundles/org.openhab.binding.energidataservice/README.md:88]: This intendation level feels wrong.
I would rather keep the first { after the arrow and intend the code of the callback one more.

Comment (jlaur) [reply bundles/org.openhab.binding.energidataservice/README.md]: Thanks, I hope it's better now. A few tabs had also crept in. I pasted it into https://beautifier.io/ to verify.

File: bundles/org.openhab.binding.energidataservice/README.md [hunk 2]
```diff
@@ -69,6 +71,41 @@ This has the following advantages:
...
+```javascript
+rules.when()
+    .channel('energidataservice:service:energidataservice:electricity#event').triggered('DAY_AHEAD_AVAILABLE')
+    .then(event => {
+        var timeSeries = new items.TimeSeries('REPLACE');
+        var start = time.LocalDate.now().atStartOfDay().atZone(time.ZoneId.systemDefault());
+        var spotPrices = items.SpotPrice.persistence.getAllStatesBetween(start, start.plusDays(2));
+        for (var spotPrice of spotPrices) {
+            var totalPrice = spotPrice.quantityState
+                .add(items.GridTariff.persistence.persistedState(spotPrice.timestamp).quantityState)
```
Comment (florian-h05) [inline bundles/org.openhab.binding.energidataservice/README.md:93]: Looks good now!

Just one minor remark:

```suggestion
                .add(items.GridTariff.persistence.persistedState(spotPrice.timestamp))
```

Should work as well.

File: bundles/org.openhab.binding.energidataservice/README.md [hunk 3]
```diff
@@ -69,6 +71,41 @@ This has the following advantages:
...
+##### Time Series
+
+Group items with aggregate functions are not automatically recalculated into the future when time series for child items are updated.
+Therefore, the `SUM` function mentioned above will only work for the current price.
+Calculation of future total prices can be achieved with a rule:
+
+:::: tabs
+
+::: tab JavaScript
+
```
Comment (florian-h05) [inline bundles/org.openhab.binding.energidataservice/README.md:83]: Probably also mention that the example is for a file-based rule.

File: bundles/org.openhab.binding.energidataservice/README.md (hunk 2 above)
Comment (jlaur) [reply bundles/org.openhab.binding.energidataservice/README.md:93]: Hmm, how? `Item.persistence.persistedState(timestamp)` returns `PersistedItem` according to documentation. `PersistedItem` contains the properties `state`, `numericState`, `quantityState`, `rawState` and `timestamp`. I need a `Quantity` to add it to `spotPrice.quantityState` which is also `Quantity`. Still, maybe I'm missing something, so I tried with this result:

```
org.graalvm.polyglot.PolyglotException: TypeError: Argument of wrong type provided, required Item, string or Quantity.
```

File: bundles/org.openhab.binding.energidataservice/README.md (hunk 3 above)
Comment (jlaur) [reply bundles/org.openhab.binding.energidataservice/README.md:83]: Done.

File: bundles/org.openhab.binding.energidataservice/README.md (hunk 2 above)
Comment (florian-h05) [reply bundles/org.openhab.binding.energidataservice/README.md:93]: Oh sorry, I thought that you got a normal Item there (probably too late to play around with code now).

**Review by florian-h05 — APPROVED** [review]

**Review by lsiepel — APPROVED** [review]
LGTM, thanks

**Pull request merged**